### PR TITLE
Add support for applying graphql dml

### PR DESCRIPTION
**Pull request by erlendvollset**
- Add support for applying graphql DML in data modeling client
- Update changelog and version

## Description
Please describe the change you have made.

## Checklist:
- [ ] Tests added/updated.
- [ ] Documentation updated. Documentation is generated from docstrings - these must be updated according to your change.
  If a new method has been added it should be referenced in [cognite.rst](https://github.com/cognitedata/cognite-sdk-python/blob/master/docs/source/cognite.rst) in order to generate docs based on its docstring.
- [ ] Changelog updated in [CHANGELOG.md](https://github.com/cognitedata/cognite-sdk-python/blob/master/CHANGELOG.md).
- [ ] Version bumped. If triggering a new release is desired, bump the version number in [_version.py](https://github.com/cognitedata/cognite-sdk-python/blob/master/cognite/client/_version.py) and [pyproject.toml](https://github.com/cognitedata/cognite-sdk-python/blob/master/pyproject.toml) per [semantic versioning](https://semver.org/).

### CODE REVIEW

File: cognite/client/_api/data_modeling/graphql.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+                ) {{
+                    result {{
+                        graphQlDml
+                        externalId
+                        version
+                    }}
+                }}
+            }}
+        """
+        self._post_graphql(url_path="/dml/graphql", query=graphql_body)
```
Comment (doctrino) [inline cognite/client/_api/data_modeling/graphql.py:44]: Doesn't this create a response?

File: cognite/client/_api/data_modeling/graphql.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+            dml (str): The DML to apply.
+            previous_version (Optional[str]): The previous version of the DML. Specify to reuse view versions from
+                previous data model version.
+        """
+        data_model_id = DataModelId.load(id)
+        graphql_body = f"""
+            mutation UpsertGraphQlDmlVersion{{
+                upsertGraphQlDmlVersion(
+                    graphQlDmlVersion: {{
+                        space: "{data_model_id.space}",
```
Comment (ebrattli) [inline cognite/client/_api/data_modeling/graphql.py:29]: We can add `name` and `description` as fields here as well

**Review by doctrino — COMMENTED** [review]
Only wonder if the `apply_dml` method should return something?

File: cognite/client/_api/data_modeling/graphql.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+                        externalId: "{data_model_id.external_id}",
+                        version: "{data_model_id.version}",
+                        graphQlDml: "{dml}",
+                        previousVersion: "{previous_version}"
+                    }}
+                ) {{
+                    result {{
```
Comment (ebrattli) [inline cognite/client/_api/data_modeling/graphql.py:36]: We should query for errors here so the user can see the errors. More result fields are available if we want to provide them through the SDK.

Comment (erlendvollset) [reply cognite/client/_api/data_modeling/graphql.py:49]: Hmm. We are getting the errors even though we're not specifying them here.. See `test_apply_dml_invalid`.

File: cognite/client/_api/data_modeling/graphql.py (hunk 2 above)
Comment (erlendvollset) [reply cognite/client/_api/data_modeling/graphql.py:40]: Done.

File: cognite/client/_api/data_modeling/graphql.py (hunk 1 above)
Comment (erlendvollset) [reply cognite/client/_api/data_modeling/graphql.py]: Made apply_dml return a DataModelId now.

File: cognite/client/_api/data_modeling/graphql.py [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+        previous_version: Optional[str] = None,
+    ) -> DataModelId:
+        """Apply the DML for a given data model.
+
+        Args:
+            id (DataModelIdentifier): The data model to apply DML to.
+            dml (str): The DML to apply.
+            previous_version (Optional[str]): The previous version of the data model. Specify to reuse view versions from
+                previous data model version.
+        """
```
Comment (doctrino) [inline cognite/client/_api/data_modeling/graphql.py:34]:
```suggestion
            previous_version (Optional[str]): The previous version of the data model. Specify to reuse view versions from
                previous data model version.
         Returns:
             The DataModelID of the newly applied data model. 
        """
```

**Review by doctrino — APPROVED** [review]
Looks good to me, only document the return type.

**Pull request merged**